mfr/providers/osf/provider.py

```diff
@@ -73,7 +73,7 @@ def __init__(self, request, url, action=None):
             metadata_response = await self._make_request(
                 'HEAD',
                 download_url,
-                headers={settings.MFR_ACTION_HEADER: self.action or ''}
+                headers=({settings.MFR_ACTION_HEADER: self.action} if self.action else None)
```

Comment:
* I don't think you need the extra `()`. This should be good.

```python
headers={settings.MFR_ACTION_HEADER: self.action} if self.action else None
```

* A header with empty value is different from not having the header. [Here](https://github.com/CenterForOpenScience/waterbutler/blob/develop/waterbutler/auth/osf/handler.py#L108) is how WB uses this header. It shouldn't be sent if it is empty. 👍 

```python
...
mfr_action = request.headers[settings.MFR_ACTION_HEADER].lower()
try:
    osf_action = mfr_action_map[mfr_action]
...
```